crates/ty_python_semantic/resources/mdtest/snapshots/union_call.md_-_Calling_a_union_of_function_types_-_A_smaller_scale_example.snap

```diff
@@ -0,0 +1,70 @@
+---
+source: crates/ty_test/src/lib.rs
+expression: snapshot
+---
+---
+mdtest name: union_call.md - Calling a union of function types - A smaller scale example
+mdtest path: crates/ty_python_semantic/resources/mdtest/diagnostics/union_call.md
+---
+
+# Python source files
+
+## mdtest_snippet.py
+
+```
+ 1 | def f1() -> int:
+ 2 |     return 0
+ 3 | 
+ 4 | def f2(name: str) -> int:
+ 5 |     return 0
+ 6 | 
+ 7 | def _(flag: bool):
+ 8 |     if flag:
+ 9 |         f = f1
+10 |     else:
+11 |         f = f2
+12 |     # error: [too-many-positional-arguments]
+13 |     # error: [invalid-argument-type]
+14 |     x = f(3)
+```
+
+# Diagnostics
+
+```
+error: lint:invalid-argument-type: Argument to this function is incorrect
+  --> src/mdtest_snippet.py:14:11
+   |
+12 |     # error: [too-many-positional-arguments]
+13 |     # error: [invalid-argument-type]
+14 |     x = f(3)
+   |           ^ Expected `str`, found `Literal[3]`
+   |
+info: Function defined here
+ --> src/mdtest_snippet.py:4:5
+  |
+2 |     return 0
+3 |
+4 | def f2(name: str) -> int:
+  |     ^^ --------- Parameter declared here
+5 |     return 0
+  |
+info: Union variant `def f2(name: str) -> int` is incompatible with this call site
+info: Union type `(def f1() -> int) | (def f2(name: str) -> int)` is not callable
```

Comment:
I think I mentioned this in a comment on the previous PR: I think we should reserve "not callable" to describe "objects that don't support the Python calling protocol" (that's what the `call-not-callable` diagnostic describes). So I would rather give this context in a way that doesn't use the phrase "not callable". One option could be "Attempted to call union type `...`" or "Error calling union type `...`"

Reply:
Ah right, thanks for repeating that here.

I went with your "Attempt to call union type `...`" verbiage here. I think it's still not great, but I get want to keep the "not callable" verbiage reserved.